Detail section

RileyManda opened the pull request:
I have completed the following new features Detail View and Javascript objects development for the Mobile and Desktop version of the Portfolio Project according to the stipulated [ProjectRequirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md).

The following work has been implemented on the mobile and desktop versions of the portfolio:

- JavaScript to manipulate DOM elements.
- Implemented JavaScript events.
- Use objects to store and access data.
- When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
- When the user clicks (or taps) the close (X) button, the popup disappears.
- Update the main page so that the projects section is created dynamically using information stored in JavaScript objects.
- Pop up details view.

In addition, I have ensured that :

- There are no linter errors in the project.
- Used Correct git-flow.
- Documented my work in a professional manner according to [Professional code documenting requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
- Included a descriptive [Readme](https://github.com/RileyManda/portfolio-website/blob/desktop-version/README.md) file.
- Used descriptive commit messages
- Followed the recommended HTML&CSS best [practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md)
- Included an attribution to my Pair-Programming partner for working on the detail view styling.

Please review the changes and let me know if there are any issues or suggestions for improvement. I look forward to your feedback. 

Thank you! 
Best regards,🙏

BenMukebo commented on index.html:
```diff
@@ -20,29 +20,67 @@
+    <!-- mobile-menu END-->
+    <!-- Details view -->
+    <div class="detail-view hidden" id="detail-view">
+    <div class="detail-view-container">
+        <button onclick="closeView()" class="detail-close-btn" id="close-view">
+            <!-- <i class="fas fa-times desktop-close"></i> -->
+            </button>
+           
+        <div class="image-container"></div>
+        <h2 class="detail-title" id="detail-title"></h2>
+<div class="detail-tags">
+    <ul>
+        <li>Ruby on Rails</li>
+        <li>CSS</li>
+        <li>Javascript</li>
+    </ul>
+</div>
+<div class="detail-summary">
+
+<p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's
+    standar dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s.</p>
+<p>
+<p>Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's
+    standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it
+    1960s.</p>
+</div>
+
+
+     <div class="detail-btn-container">
+<button class="detail-btn">See Live<a href="https://rileymanda.github.io/portfolio-website/"><img src="./assets/icons/live-icon.svg"
+            alt="Twitter icon"></a></button>
+<button class="detail-btn">See Source<a href="https://github.com/RileyManda/portfolio-website"><img src="./assets/icons/github-light.svg" alt="Twitter icon"></a></button>
+     </div>
+        
```
- [ ] In this particular project, the section of the whole project along with the models should be created dynamically with Javascript. I can see that the base projects section was populated with the data from the `workcards` array good job! 👏🏻 , but the Modal popup data section is still not being created with Javascript.  Please use Javascript to dynamically render your Modal popup windows data as you did with the projects section.

- That way, adding projects in the future will be easy. You will only need to add it in one place - as another object in the `projectList` array. 😉 👍🏻 
 
 - (Note: the works section must be removed from HTML and generated with javascript )

BenMukebo commented on works-cards.js:
```diff
@@ -0,0 +1,32 @@
+import { workcards } from './data.js';
+
+const generateworkCardHTML = (workCard) => {
+  const { title, subtitle, tags } = workCard;
+  const tagsHTML = tags.map((tag) => `<li>${tag}</li>`).join('');
+  const cardHTML = `
+    <div class="works-card">
+      <div class="works-card-content">
+        <h3 class="works-card-title">${title}<span><br>${subtitle}</span></h3>
+        <div class="works-tags">
+          <ul>
+            ${tagsHTML}
+          </ul>
+        </div>
+        <button class="project-btn" onclick="openDetailView()">See Project</button>
+      </div>
+    </div>
+  `;
```
- [ ] Your popup must follow the design for both desktop and mobile versions from [the chosen template in Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1) (e.g., font, colours, images, tests, margins,...)  kindly make this requested changes: 


- Please blur the sub background when the popup is open, it will make the popup more like Figma 👍🏿
- It would be better if the background is not scrollable when the popup page is open. you can use the `overflow:hidden;` CSS property to avoid this. 😄 
- Make sure the max width of your popup windows is 920px as shown on the Figma design.
- You need to respect the text description font size, line-height, font family, .... and space padding or margin ex: between the 'Tech stack` (HTML, CSS, js, ...). `text description` and  the  `See live and See source` buttons

Your Implementation | Figma Design
:---:|:---:
![image](https://github.com/RileyManda/portfolio-website/assets/79058364/e67da3d4-7753-46fc-a3f9-d87a49095557) | ![181](https://user-images.githubusercontent.com/79058364/189106086-ad46721f-211a-41e9-97b7-421fb32231c2.png) 


- [ ]  Please do the same for the desktop size 👨🏻‍🎨 🎨 

- Actually, I don't see the full height of your page popups when I'm on the desktop size, please make sure 100% height is shown 👨🏻‍🎨

Your Implementation | Figma Design
:---:|:---:
![image](https://github.com/RileyManda/portfolio-website/assets/79058364/fab4328e-d02b-47e4-9983-2e00a70c07a3) | ![18](https://user-images.githubusercontent.com/79058364/189106145-f947cebe-3a5a-45a8-9851-ada6c6544c46.png)

BenMukebo reviewed (changes requested):
### Hi @RileyManda,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- No linter errors. ✔️ 
- GitHub flow was correctly used.  ✔️   
- The [JavaScript](https://github.com/microverseinc/linters-config/tree/master/javascript) and [HTML&CSS](https://github.com/microverseinc/linters-config/tree/master/html-css) linters is set correctly  ✔️ 
-  When the user clicks the button to check project details, the popup with details about the project appears.✔️   
- When the user clicks the close button, the popup disappears ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda commented:
I have completed the following modifications on the features Detail View and Javascript objects development for the Mobile and Desktop version of the Portfolio Project as requested by the reviewer.

The following changes were made to the project:

- Modals have been created dynamically with Javascript.
- Popup view follows the design for both desktop and mobile versions.
- Grey overlay has been implemented when popup view is visible
- Desktop and Mobile version title,tags and buttons have been adjusted to match the figma design
- overflow:hidden; has been implemented to prevent background scrolling

I have developed the  to adhere to the requirements stipulated in the development [requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md) by:

- Implementing JavaScript events.
- Using objects to store and access data.
- When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
- When the user clicks (or taps) the close (X) button, the popup disappears.
- Update the main page so that the projects section is created dynamically using information stored in JavaScript objects.
- Pop up details view.

In addition, I have ensured that :

There are no linter errors in the project.
Used Correct git-flow.
Documented my work in a professional manner according to [Professional code documenting requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
Included a descriptive [Readme](https://github.com/RileyManda/portfolio-website/blob/desktop-version/README.md) file.
Used descriptive commit messages
Followed the recommended HTML&CSS best [practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md)
Included an attribution to my Pair-Programming partner for working on the detail view styling.
Please review the changes and let me know if there are any issues or suggestions for improvement. I look forward to your feedback. 

Thank you! 
Best regards,🙏

franclobo reviewed (commented):
Hi @RileyManda, 

![terminator](https://github.com/RileyManda/portfolio-website/assets/58642949/921897d9-7b18-4487-9c73-5d10e8cb205f)

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is a pair-programming activity and it doesn't have commits from your partner.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

RileyManda commented:
_My pair-coding partner has contributed to the project:To render the project **valid** as it was marked invalid by the previous reviewer_

The following modifications on this feature branch were made  as requested by the previous reviewer:

- Modals have been created dynamically with Javascript.
- Popup view follows the design for both desktop and mobile versions.
- Grey overlay has been implemented when popup view is visible
- Desktop and Mobile version title,tags and buttons have been adjusted to match the figma design
- overflow:hidden; has been implemented to prevent background scrolling

I have also developed the project to adhere to the requirements stipulated in the development [requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md) by:

- Implementing JavaScript events.
- Using objects to store and access data.
- When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
- When the user clicks (or taps) the close (X) button, the popup disappears.
- Updated the main page so that the projects section is created dynamically using information stored in JavaScript objects.
- Implemented Pop up details view design.
- Incorporated a live link to the project and view source to source code in the modal view buttons.

In addition, I have ensured that :

- There are no linter errors in the project.
- Used Correct git-flow.
- Documented my work in a professional manner according to [Professional code documenting requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
- Included a descriptive [Readme](https://github.com/RileyManda/portfolio-website/blob/desktop-version/README.md) file.
- Used descriptive commit messages
- Followed the recommended HTML&CSS best [practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md)
- Followed the recommended Javascript best [practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)

Please review the changes and let me know if there are any issues or suggestions for improvement. I look forward to your feedback. 

Thank you! 
Best regards,🙏

Davidosky007 reviewed (commented):
Hi @RileyManda 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't show contribution from your coding partner, if your partner contributed code to this project it will be visible in your commit history for this project but currently i can only see your commits  

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

RileyManda commented:
You are marking a project as invalid yet my coding partner has contributed to the feature. Please kindly check again. @Davidosky007

Davidosky007 commented:
> You are marking a project as invalid but my coding partner has contributed to the feature. Please kindly check again. @Davidosky007 
> 
> <img width="937" alt="Screen Shot 2023-05-26 at 8 20 15 AM" src="https://github.com/RileyManda/portfolio-website/assets/37291194/c07b4c2c-b68c-4b49-a52d-32c232a51c2b">
> 
> 

I meant on this particular milestone portfolio details pop up window and not the whole repository In which this PR was raised , I can only see your contributions

RileyManda commented:
@Davidosky007
Would you recommend I open a new PR? Because my pair-coding partner contributed to this feature branch. I do not understand why it's not visible.

Davidosky007 commented:
> @Davidosky007
> 
> Would you recommend I open a new PR? Because my pair-coding partner contributed to this feature branch. I do not understand why it's not visible.

Maybe it's because you have merged the PR , you can try that out and add a link to this PR as a reference with an explanation for why you open another PR
